### DISCUSSION TITLE
Manually set MJPEG resolution

### DISCUSSION THREAD

**Discussion by DomiStyle** [blakeblackshear/frigate#1267] [General]
Hey there,

I have 3 2160p cameras in my Frigate setup right now and everything seems to be working perfectly fine so far.

My only issue is with the quality of the MJPEG streams. It seems like it uses the canvas height to scale the image automatically, probably to save resources.

Especially on mobile the image quality is abysmal, both in the web interface and in the Home Assistant app.
On desktop it's slightly better but still miles away in terms of quality from a full 2160p image.

Is it possible to increase the minimum size?

**Comment by blakeblackshear** [comment]
It is probably the resize algorithm on the back end. It uses a linear resize because it is much faster, but it results in low quality when downscaling significantly. There isn't a way to change the resize algorithm or change the requested resolution from the frontend. The next version will use jsmpeg for live view and may make this better.

**Reply by DomiStyle** [reply]
Sounds good, does the new version affect Home Assistant too or will it still use MJPEG?

**Reply by blakeblackshear** [reply]
Homeassistant should be using the RTMP feed.

**Reply by DomiStyle** [reply]
I assume with RTMP disabled in Frigate it still uses the MJPEG feed?

**Reply by blakeblackshear** [reply]
Yes.

**Comment by DomiStyle** [comment]
In case anyone else runs into this issue, I worked around the limitation in my nginx config:

```
set $is_latest "";

if ($uri ~ (.+)/latest.jpg)
{
    set $is_latest "L";
}

if ($args ~* (.*)(h=[0-9]+)(.*))
{
    set $is_latest "${is_latest}R";
}

if ($is_latest = LR)
{
    set $args $1h=2160$3;
}
```

Add above `location /api/` in Frigate nginx config or in reverse proxy config. Replace 2160 with any resolution you need.

This will scale any image to 2160p, no matter what the requested height is. Not ideal but good enough for now.

Works in Home Assistant and on desktop, doesn't work on Firefox for Android but that's probably a limitation of the canvas there.
